corehq/apps/users/bulkupload.py

```diff
@@ -372,6 +426,13 @@ def is_password(password):
                     if email:
                         user.email = email
                     user.save()
+                    if location_code:
+                        loc = location_cache.get(location_code)
+                        if user.location_id != loc._id:
+                            # this triggers a second user save so
+                            # we want to avoid doing it if it isn't
+                            # needed
+                            user.set_location(loc)
```

Comment:
Ugh no this was on purpose. When setting that delegate case, the user needs to be a saved user. So doing only one save would crash on creation.